File: projects/integration/transforms-python/src/python/exposure/current/dl130_market_risk_stage_result_summary.py
# ===== import: python function

# ===== import: palantir functions
from transforms.api import transform_df, Input, Output, configure

# ===== import: our functions
from python.util.bp_dateutils import filter_latest_runs
from python.util.bp_config import Config
from python.util.bp_constants import LOOK_BACK_PERIOD_CONFIG
from python.bp_flush_control import DL130_FLUSH, DL130_FLUSH_START_DATE, DL130_PROFILE


def convert(df):
    config = Config(look_back_period=LOOK_BACK_PERIOD_CONFIG,
                    flush=DL130_FLUSH,
                    flush_start_date=DL130_FLUSH_START_DATE)
    df = df.filter((df.Analysis_Date >= config.start_date) & (df.Analysis_Date <= config.end_date))
    df = filter_latest_runs(df, 'Analysis_Date', 'Run_Datetime')
    return df


@configure(profile=DL130_PROFILE)
@transform_df(
    Output("/BP/IST-IG-DD/data/technical/exposure/current/endur/dl130_market_risk_stage_result_summary"),
    df=Input("/BP/IST-IG-DD/data/technical/exposure/typed/endur/dl130_market_risk_stage_result_summary"),
)
def my_compute_function(df):
    return convert(df)

File: projects/IST-IG-DD-JAVA/transforms-java/src/main/java/myproject/datasets/exposure/current/dl130_market_risk_stage_result_summary.java
package myproject.datasets.exposure.current;
import myproject.datasets.util.BP_DateUtils;
import org.apache.spark.sql.Column;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;

import com.palantir.transforms.lang.java.api.Compute;
import com.palantir.transforms.lang.java.api.Input;
import com.palantir.transforms.lang.java.api.Output;

import java.util.Date;

public class dl130_market_risk_stage_result_summary {

    public Dataset<Row> convert(Dataset<Row> df) {
        Date[] dates = BP_DateUtils.getStartEndDate();
        df = BP_DateUtils.filterLatest(df, dates[0], dates[1], "Analysis_Date", "Run_Datetime");
        return df;
    }

    @Compute
    @Output("/BP/IST-IG-DD/data/technical/exposure_java/current/endur/dl130_market_risk_stage_result_summary")
    public Dataset<Row> myComputeFunction(
            @Input("/BP/IST-IG-DD/data/technical/exposure_java/typed/endur/dl130_market_risk_stage_result_summary") Dataset<Row> df
    ){
        return convert(df);
    }

}

